app/src/main/java/io/gnosis/safe/ui/base/BaseStateViewModel.kt

```diff
@@ -16,7 +20,36 @@ data class AppDispatchers(
     val background: CoroutineDispatcher = Dispatchers.IO
 )
 
-abstract class BaseStateViewModel<T>(private val dispatchers: AppDispatchers) : ViewModel() where T : BaseStateViewModel.State {
+abstract class BaseStateViewModel<T>(private val dispatchers: AppDispatchers, private val connectivityManager: ConnectivityManager? = null) :
+    ViewModel() where T : BaseStateViewModel.State {
+
+    private lateinit var networkCallback: ConnectivityManager.NetworkCallback
```

Comment:
Couldn't we add our own wrapper to `ConnectivityManager`, something like a `ConnectivityMonitor` or `ConnectivityObserver` that we consume in the `MainActivity`? Or even everywhere, that doesn't matter so much. There are times that we would like to have `viewModels` that aren't necessarily `BaseStateViewModel` which have simpler logic, but may need to also monitor the network state. This is again an argument against this sort of code coupling.

Reply:
not every screen needs connectivity updates, so I would not place observers in MainActivity. Also, it is fine to not provide connectivity manager and there will be no updates. Adding wrapper will not change much - it will still be connectivity manager registering and unregistering network callback.

Reply:
It would add the possibility of of providing this information as `flow` or somehow that allows subscription based consumption. Then that is easily couple-able with screen lifecycles. The network callback will be registered once and then consumers can subscribe to its changes. Then we can isolate this responsibility in a single dependency.